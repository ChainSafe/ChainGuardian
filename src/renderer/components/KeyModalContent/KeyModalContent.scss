.key-input-container {
    display: flex;
    flex-direction: column;
    width: 100%;

    .inputform {
        width: 100%;
        box-sizing: border-box;
    }
}

.info-container {
    display: flex;
    flex-wrap: nowrap;

    div:first-child {
        width: 10%;
        margin-right: 5%;
    }

    div:last-child {
        width: 85%;
    }
}

.mt-32 {
    margin-top: 32px;
}

.submit-button-container {
    margin-top: 40px;
    text-align: center;
}