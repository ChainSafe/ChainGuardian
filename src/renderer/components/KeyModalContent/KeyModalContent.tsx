import React, {useState, ReactElement, useEffect} from "react";
import {InputForm} from "../Input/InputForm";
import {ButtonPrimary} from "../Button/ButtonStandard";
import {isValidMnemonicOrPrivateKey, isValidPublicKey} from "../../services/utils/input-utils";

interface IKeyModalProps {
    title: string,
    description?: string,
    onSubmit: (input: string) => void,
    placeholder: string,
    signing: boolean
}

export default function KeyModalContent(props: IKeyModalProps): ReactElement {
    const [input, setinput] = useState("");
    const [valid, setvalid] = useState(false);
    const [errorMessage, setErrorMessage] = useState("");

    const handleChange = (e: React.FormEvent<HTMLInputElement>): void => {
        const input = e.currentTarget.value;
        setinput(input);

        if(input === ""){
            setvalid(false);
            setErrorMessage("");
            return;
        }

        const {isValid, message} = props.signing ? isValidMnemonicOrPrivateKey(input) : isValidPublicKey(input);
        setvalid(isValid);

        if(!isValid){
            setErrorMessage(message);
        }
    };

    const handleSubmit = (): void => {
        if(valid){
            props.onSubmit(input);
        }
    };

    useEffect(() => {}, [valid]);

    return (
        <>
            <h1>{props.title}</h1>
            {props.description &&
                <p>
                    {props.description}
                </p>
            }
            <div className={`key-input-container ${props.description ? "" : "mt-32"}`}>
                <InputForm
                    inputId="inputKey"
                    focused
                    onChange={handleChange}
                    inputValue={input}
                    valid={errorMessage === "" && valid === false ? undefined : valid}
                    errorMessage={errorMessage}
                    placeholder={props.placeholder} />
                <span className="submit-button-container">
                    <ButtonPrimary buttonId="submit" disabled={!valid} onClick={handleSubmit}>Submit</ButtonPrimary>
                </span>
            </div>
        </>
    );
}
