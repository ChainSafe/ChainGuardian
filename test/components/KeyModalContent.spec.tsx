import * as React from "react";
import * as renderer from "react-test-renderer";
import KeyModalContent from "../../src/renderer/components/KeyModalContent/KeyModalContent";

describe("KeyModalContent", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(
                <KeyModalContent
                    title='Test'
                    onSubmit={(): void => {}}
                    validatorIndex='1'
                    placeholder='Test placeholder'></KeyModalContent>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders description", () => {
        const tree = renderer
            .create(
                <KeyModalContent
                    title='Test'
                    validatorIndex='1'
                    onSubmit={(): void => {}}
                    description='Test description'
                    placeholder='Test placeholder'
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
