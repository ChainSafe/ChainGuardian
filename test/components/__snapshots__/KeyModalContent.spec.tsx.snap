// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyModalContent renders correctly 1`] = `
Array [
  <h1>
    Test
  </h1>,
  <div
    className="key-input-container mt-32"
  >
    <form>
      <div
        className="label"
      />
      <input
        autoFocus={true}
        className="inputform "
        id="inputKey"
        onChange={[Function]}
        placeholder="Test placeholder"
        value=""
      />
      <div
        className="error-message"
      />
    </form>
    <span
      className="submit-button-container"
    >
      <button
        className="button btn-primary "
        disabled={true}
        id="submit"
        onClick={[Function]}
      >
        <div
          className=""
        >
          Submit
        </div>
      </button>
    </span>
  </div>,
]
`;

exports[`KeyModalContent renders description 1`] = `
Array [
  <h1>
    Test
  </h1>,
  <p>
    Test description
  </p>,
  <div
    className="key-input-container "
  >
    <form>
      <div
        className="label"
      />
      <input
        autoFocus={true}
        className="inputform "
        id="inputKey"
        onChange={[Function]}
        placeholder="Test placeholder"
        value=""
      />
      <div
        className="error-message"
      />
    </form>
    <span
      className="submit-button-container"
    >
      <button
        className="button btn-primary "
        disabled={true}
        id="submit"
        onClick={[Function]}
      >
        <div
          className=""
        >
          Submit
        </div>
      </button>
    </span>
  </div>,
]
`;
